File: 401-600/e589.java
import java.util.ArrayList;
import java.util.List;

class e589 {
    
    class Node {
        public int val;
        public List<Node> children;
    
        public Node() {}
    
        public Node(int _val,List<Node> _children) {
            val = _val;
            children = _children;
        }
    }

    // recursive approach.
    public List<Integer> preorder(Node root){
        List<Integer> ans = new ArrayList<>();
        preorderhelper(root, ans);
        return ans;
    }

    public void preorderhelper(Node root, List<Integer> sol) {
        if (root == null){
            return;
        } 
        
        sol.add(root.val);
        for (Node child: root.children){
            preorderhelper(child, sol);
        }
    }
}

File: 401-600/e589.py
# Definition for a Node.
class Node(object):
    def __init__(self, val, children):
        self.val = val
        self.children = children

class Solution(object):
    def preorder(self, root):
        """
        :type root: Node
        :rtype: List[int]
        """

        if not root:
            return visited

        visited = []
        stack = [root]

        while stack:
            current = stack.pop(0)
            if current:
                visited.append(current.val)
                stack = current.children + stack
        return visited